jetstream/kv.go

```diff
@@ -174,10 +174,17 @@ type (
 		// Deprecated: Use ListKeys instead to avoid memory issues.
 		Keys(ctx context.Context, opts ...WatchOpt) ([]string, error)
 
+		// KeysWithFilters returns a filtered list of keys in the bucket.
+		// It returns a complete slice of matching keys.
+		KeysWithFilters(ctx context.Context, kv KeyValue, filters ...string) ([]string, error)
+
 		// ListKeys will return KeyLister, allowing to retrieve all keys from
 		// the key value store in a streaming fashion (on a channel).
 		ListKeys(ctx context.Context, opts ...WatchOpt) (KeyLister, error)
 
+		// ListKeysWithFilters returns a KeyLister for filtered keys in the bucket.
+		ListKeysWithFilters(ctx context.Context, kv KeyValue, filters ...string) (<-chan string, error)
```

Comment:
This method should return `KeyLister` as noted in the previous comment:
```go
ListKeysFiltered(ctx context.Context, filters ...string) (KeyLister, error)
```

Also, the `kv KeyValue` parameter is not necessary.